libsmtutil/CHCSolverInterface.h

```diff
@@ -54,8 +54,8 @@ class CHCSolverInterface
 	};
 
 	/// Takes a function application _expr and checks for reachability.
-	/// @returns solving result and a counterexample graph, if possible.
-	virtual std::pair<CheckResult, CexGraph> query(
+	/// @returns solving result, an invariant, and counterexample graph, if possible.
+	virtual std::tuple<CheckResult, Expression, CexGraph> query(
```

Comment:
Does it make sense to have structure instead of returning tuple?

Reply:
In this case I prefer a `tuple` because it's less verbose and still clear enough imo.